src/useFormState.ts

```diff
@@ -47,18 +48,31 @@ export type UseFormStateOpts<T, I> = {
    *
    * Does not need to be stable/useMemo'd.
    */
-  autoSave?: (state: ObjectState<T>) => Promise<void>;
+  autoSave?: (state: ObjectState<T>) => Promise<unknown>;
 };
 
 // If the user's autoSave hook makes some last-minute `.set` calls to sneak
 // in some business logic right before their GraphQL mutation call, ignore it
 // so that we don't infinite loop.
-let isAutoSaving = false;
+let isAutoSaving: "queued" | "in-flight" | false = false;
 
 // `pendingAutoSave` is a flag for determining if we need to immediately call `maybeAutoSave` again after the initial Promise finishes
 // This could happen if a field triggers auto-save while another field's auto-save is already in progress.
 let pendingAutoSave = false;
 
+let timerId: any;
+if ("afterEach" in global) {
```

Comment:
This is too bad to have test specific logic in our production code.

Reply:
Yeah, I was really getting burned by cross-test leaks/breakages when I wrote this, but I think some of the `runOnlyPendingTimers` / `getTimerCount` tricks we've learned has made this less likely to happen, so yeah I've removed it.